mlutze/flix
# CLI-GENERATED-tvar-sym-lsp

- fix: improve performance of `OccurenceAnalyzer` (#3262)
- feat: introduce `ScopedArray` (#3260)
- chore: handle `LetRegion` correctly in `Redundancy` (#3264)
- chore: handle `LetRegion` in LSP `Indexer`. (#3267)
- feat: ensure that region variables do not escape their scope (#3273)
- refactor: remove `DemandList` (#3274)
- feat: add `foreach` to `DelayList` (#3265)
- refactor: prepare `DelayList` release (#3243)
- feat: add alternative region syntax (#3279)
- refactor: use regions in type rules for arrays. (#3281)
- feat: add `reg` type constructor (#3283)
- refactor: use let rec in `Map` (#3287)
- fix: encode DocumentSymbol into JSON properly (#3289)
- refactor: add optimizer pipeline (#3240)
- test: add tests for incremental compilation (#3286)
- Refactor: `StringBuilder` to subject-last (#3222)
- feat: add greetings.yml (#3293)
- chore: report phase times for `Inliner`, `OccurrenceAnalyzer`, and `Reducer`. (#3295)
- refactor: improve resource management in packager (#3285)
- refactor: Use `foreach` in cases using `foldLeft` (#3296)
- refactor: purity in SimplifiedAst (#3291)
- feat: add foldRightLazy (#2896)
- refactor: use higher precision in `Timer.sleepAndSend` (#3299)
- fix: add `WeederError.ReservedName` (#3269)
- refactor: use let rec in `List` (#3275)
- refactor: add `Purity` to `LiftedAst` and `OccurenceAst` (#3304)
- refactor: use let rec in `MutList` (#3292)
- refactor: StringBuilder to subject-last (#3252)
- refactor: use let rec in `Iterator` (#3284)
- feat: add `Region` kind as an alias for `Bool` kind. (#3311)
- refactor: use let rec in `Nel` (#3320)
- refactor: use let rec in `Option` (#3321)
- chore: kill the Terminator (#3318)
- refactor: use let rec in `Result` (#3324)
- feat: allow annotations on enums (#3290)
- refactor: use let rec in `Set` (#3326)
- fix: `INVOKESTATIC` to work for both classes and interfaces (#3331)
- feat: erase `LetRegion` in `Lowering`. (#3340)
- feat: simplify rigid variables (#3341)
- chore: remove old region syntax. (#3342)
- feat: add `Newable` type class (#3344)
- chore: remove rigid variables. (#3347)
- chore: rename `Region` to `Scope`. (#3348)
- feat: introduce syntax for allocating arrays in regions. (#3349)
- refactor: remove `mut` keyword (#3350)
- refactor: use let rec in `Validation` (#3329)
- refactor: remove extra newline in error messages (#2940)
- fix: include definition site in references (#3343)
- feat: add `region` expression (#3352)
- fix: bug in `Kinder` (#3358)
- chore: remove scopedness-related code (#3357)
- refactor: use `ManyBranches` in `OccurrenceAnalyzer`. (#3298)
- feat: add `Scoped` type class (#3360)
- feat: remove `System.exit` call in `Main.class` and support all return types of the flix main (#3364)
- fix: use def sites in `FindReferencesProvider` (#3353)
- feat: `File` - Milestone 4 (#3241)
- feat: allow annotations on classes (#3294)
- refactor: use let rec in `String` (#3323)
- refactor: OccurrenceAnalyzer (#3373)
- feat: release 0.27.0 (#3374)
- fix: a bug in `Documentor`. (#3375)
- feat: add regions to `ArrayLit` and `ArrayNew` (#3372)
- refactor: use `ScopedRef` (regions) in `MutSet` (#3319)
- feat: type check arrays with regions (#3379)
- feat: add new syntax for `Newable.new`. (#3380)
- feat: tweak syntax of `new`. (#3381)
- feat: add syntax for `Read(...)` and `Write(...)` effects (#3385)
- feat: add support for implicit regions (#3386)
- refactor: use `Read` / `Write` syntax in MutSet (#3387)
- feat: add support for implicit regions (#3389)
- refactor: `MutSet` tests to use `new` syntax  #3391)
- feat: tweak source locations for `new` and `ref`. (#3393)
- chore: remove `reg(r)` (#3394)
- chore: improve type formatting (#2452)
- feat: add support for deprecated and experimental code hints to LSP (#3384)
- chore: fix trailing chars in `ProgressBar`. (#3403)
- feat: remove variables that are both `Dead` and `Pure` (#3396)
- chore: format Unit -> X signatures correctly (#3405)
- chore: ensure dbg package is not used in LSP (#3408)
- test: rename `Test.Exp.Reference.*` to `Test.Exp.Ref.*` (#3390)
- tests: use regions in `Test.Exp.Ref.Assign` (#3411)
- refactor: `String.split` to `Pure` (#3398)
- feat: improve type errors (#3409)
- feat: add deprecation and experimental warnings for classes (#3414)
- feat: `File` milestone 5, `move` (#3309)
- feat: add log and factorial function to Int8, Int16, and Int64 (#3413)
- feat: allow single effect without braces (#3416)
- fix: return zero in exceptional cases of `log2` and `factorial` (#3417)
- feat: add `CrashHandler` (#3419)
- feat: propagate type variable names (#3421)
- feat: Inline trivial `exps` (#3407)
- feat: `Int32.parse` and `Int64.parse` (#3423)
- feat: `File` milestone 5, `copy` (#3422)
- refactor: reorganize `Namer.visitSig` and `Namer.visitDef` (#3428)
- feat: add auto-complete of local variables and formal parameters (#3429)
- feat: ignore recursive calls when checking for uses of definitions (#3316)
- chore: prepare for new CodeLenses. (#3430)
- feat: add `List.merge` (#3427)
- chore: add explicit `entryPoint` to ASTs. (#3432)
- chore: add --entrypoint argument (#3433)
- refactor: use `--entrypoint` in CodeLenses (#3434)
- docs: add `region` example (#3397)
- refactor: `sleepAndSend` to not break on negative input (#3431)
- fix: run `main` if there is no `--entrypoint`. (#3435)
- refactor: `sleepSendAndRepeat` to not crash on negative inputs. (#3439)
- feat: add `List.splitAt` (#3438)
- fix: do not use type variable names in instantiated schemes. (#3441)
- create TypeVarSym
- use TypeVarSym in Type.Var
- fix compilation errors in tests
- add eq and hashcode to TypeVarSym
- fix formatType issue
- re-fix testing error
- WIP
- fix up some TODOs
- add rigidity and kind to TypeVarSym
- remove text, kind, and rigidity from Type.Var
- fix compilation issues
- remove this.type thingy
- fix test compilation issues
- address comments
- use tvar syms in tparams and schemese and substitutions
- add semantic token for type variable
